### PR TITLE
Fix notebook renaming

### PR DESCRIPTION
**Pull request by wnojopra**
Rename seems to be a copy and delete and operation. The copy part wasn't working because it didn't have a bucket to copy to.

### CODE REVIEW

File: src/libs/ajax.js [hunk 1]
```diff
@@ -495,8 +495,8 @@ const Buckets = signal => ({
 
       getObject,
 
-      rename: async newName => {
-        await copy(newName)
+      rename: async (newName, bucket) => {
+        await copy(newName, bucket)
```
Comment (zarsky-broad) [inline src/libs/ajax.js:499]: The difference between copy and rename is that rename presumes the bucket, so this should use `bucket` as defined above, and remain a function of 1 argument.

Thanks for catching this!

File: src/libs/ajax.js [hunk 2]
```diff
@@ -495,8 +495,8 @@ const Buckets = signal => ({
 
       getObject,
 
-      rename: async newName => {
-        await copy(newName)
+      rename: async (newName, bucket) => {
```
Comment (zarsky-broad) [inline src/libs/ajax.js]:
```suggestion
      rename: async newName => {
```

File: src/components/notebook-utils.js [hunk 3]
```diff
@@ -172,7 +172,7 @@ export const NotebookDuplicator = ajaxCaller(class NotebookDuplicator extends Co
           try {
             this.setState({ processing: true })
             await (destroyOld ?
-              Buckets.notebook(namespace, bucketName, printName).rename(newName) :
+              Buckets.notebook(namespace, bucketName, printName).rename(newName, bucketName) :
```
Comment (zarsky-broad) [inline src/components/notebook-utils.js]:
```suggestion
              Buckets.notebook(namespace, bucketName, printName).rename(newName) :
```

**Review by zarsky-broad — CHANGES_REQUESTED** [review]

File: src/libs/ajax.js (hunk 1 above)
Comment (wnojopra) [reply src/libs/ajax.js:499]: Ah, got it, makes sense. Thanks!

File: src/libs/ajax.js [hunk 4]
```diff
@@ -495,8 +495,8 @@ const Buckets = signal => ({
 
       getObject,
 
-      rename: async newName => {
-        await copy(newName)
+      rename: async (newName) => {
```
Comment (zarsky-broad) [inline src/libs/ajax.js]: Tiny thing, remove parens for single argument, thanks!

**Review by zarsky-broad — APPROVED** [review]
👍 pending minor comment

File: src/libs/ajax.js (hunk 4 above)
Comment (wnojopra) [reply src/libs/ajax.js]: Good catch, thanks.

**Comment by zarsky-broad** [comment]
You're good to merge once the CI completes! (deploy will fail because this is a fork)

**Comment by wnojopra** [comment]
Thank you, Isaac. I don't have write access to the repo so I'll actually need someone else to do the merge.

**Pull request merged**